pico/processor/src/main/java/io/helidon/pico/processor/BaseAnnotationProcessor.java

```diff
@@ -500,7 +500,7 @@ boolean isInThisModule(TypeElement type,
             }
             JavaFileObject sourceFile = path.getCompilationUnit().getSourceFile();
             Optional<Path> filePath = toPath(sourceFile.toUri());
-            Optional<Path> srcPath = toSourcePath(filePath, type);
+            Optional<Path> srcPath = filePath.isPresent() ? toSourcePath(filePath.get(), type) : Optional.empty();
```

Comment:
The following two lines can be replaced with:
```java
filePath.flatMap(it -> toSourcePath(it, type))
                    .ifPresent(services::lastKnownSourcePathBeingProcessed);
```
This is the way optional should be used

Reply:
At the cost of being less readable.

Reply:
pushed